uk.co.compucorp.civicrm.hrcore/js/src/hrcore/app.js

```diff
@@ -1,37 +1,8 @@
-/* global dataLayer */
 /* eslint-env amd */
 
-(function (window, CRM, dataLayer) {
-  define(function () {
-    'use strict';
-
-    trackContactTabVirtualPageviews();
-
-    /**
-     * Sends a virtual pageview to Google Analytics whenever the user clicks on
-     * any of the tabs of the Contact Summary page
-     *
-     * The path of the pageview is the Contact Summary page path + the selectedChild
-     * param's value of the selected tab, thus simulating the user landing
-     * directly on the tab
-     */
-    function trackContactTabVirtualPageviews () {
-      var contactPagePath;
-
-      if (!dataLayer) {
-        return;
-      }
-
-      contactPagePath = window.location.pathname + '?reset=1&cid=' + CRM.contactId;
-
-      CRM.$('#mainTabContainer').on('tabsactivate', function (event, ui) {
-        var tabName = ui.newTab[0].id.replace('tab_', '');
-
-        dataLayer.push({
-          event: 'virtual-pageview',
-          virtualPageviewPath: contactPagePath + '&selectedChild=' + tabName
-        });
-      });
-    }
-  });
-}(window, CRM, (typeof dataLayer !== 'undefined' ? dataLayer : null)));
+define([
+  'hrcore/analytics/page-tracker',
+  'hrcore/helpers/help-icon-injector'
+], function (trackContactTabVirtualPageviews) {
```

Comment:
We are injecting `'hrcore/analytics/page-tracker'`, but while using it, we are directly calling `trackContactTabVirtualPageviews `. This is a bit confusing, as it is hard to track which is function is part of which file.

To fix this, the dependency files should always return an object, so that we can do 
```
  'hrcore/analytics/page-tracker',
  'hrcore/helpers/help-icon-injector'
], function (pageTracker) {
  pageTracker.trackContactTabVirtualPageviews();
});
```

Note: When using angular, this is not a problem, because we dont have to return anything. But here as we are returning things, we should have a namespace.

Reply:
Had a discussion with @AkA84 , and the following structure seems agreeable
```
define([
  'hrcore/analytics',
  'hrcore/helpers'
], function (analytics, helpers) {
  analytics.startTrackingViews();
});
```

define([
  'hrcore/helpers'
], function (helpers) {
   helpers.appendHelpIcon();
});

Reply:
Done